packages/dashboard/src/customization/propertiesSections/components/legendDisplaySection.tsx

```diff
@@ -0,0 +1,47 @@
+import type { FC } from 'react';
+import React from 'react';
+import FormField from '@cloudscape-design/components/form-field';
+import Checkbox, {
+  CheckboxProps,
+} from '@cloudscape-design/components/checkbox';
+import { NonCancelableCustomEvent } from '@cloudscape-design/components/internal/events';
+import { ChartLegend } from '~/customization/widgets/types';
+import { dropdownConsts } from '../constants';
+
+type LegendDisplaySectionProps = {
+  visibleContent?: ChartLegend['visibleContent'];
+  onChange: (visibleContent: ChartLegend['visibleContent']) => void;
+};
+
+const { legendDisplaylist } = dropdownConsts.legendDisplay;
+
+export const LegendDisplaySection: FC<LegendDisplaySectionProps> = ({
+  visibleContent,
```

Comment:
what is this visibleContent ? if this is to show or not show the unit, shouldnt the name reflect the purpose ?

Reply:
visibleContent is to show the display options inside chartlegend obj, so i used like that. maybe I can change into legendVisibleContent in section component to make more specific.
![image](https://github.com/awslabs/iot-app-kit/assets/142866907/7460f84d-e4b3-49f6-93cb-4ae5325ab165)

Reply:
may be just content ? this also affect the dashboard API, so please a document together to share with the team about the API change @tracy-french let me know if you have thoughts here

Reply:
I think `visibleContent` makes sense. It's not the content itself in the object, it's the toggles for visibility of the content. Cloudscape uses the same naming scheme for its collection preferences (i.e., `visibleContentPreferences`): https://cloudscape.design/components/collection-preferences/.